### ISSUE TITLE
[bug] FlaskDynaconf breaks default getter of flask config

### ISSUE THREAD

**Issue by shsmad** [rochacbruno/dynaconf#521]
**Describe the bug**
In pure Flask config if you query config for nonexisting key you get `KeyError` exception.
If use FlaskDynaconf if you query config for nonexisting key you get value `None`.

Such differences break thirdparty libraries, which suppose that flask config will raise `KeyError` when queriing nonexisting key (for example, `python-social-auth`: https://github.com/python-social-auth/social-core/blob/master/social_core/strategy.py#L45 which goes to https://github.com/python-social-auth/social-app-flask/blob/master/social_flask/strategy.py#L19)

**To Reproduce**
Correct behavior:

```
from flask import Flask

app = Flask(__name__)
print("NON_EXISTING" in app.config)
print(app.config["NON_EXISTING"])
```

will give you

```
False
Traceback (most recent call last):
  File "dc_poc/1_no_dynaconf.py", line 6, in <module>
    print(app.config["NON_EXISTING"])
KeyError: 'NON_EXISTING'
```

Incorrect behavior:

```
from flask import Flask
from dynaconf import FlaskDynaconf

app = Flask(__name__)
FlaskDynaconf(app=app)
print("NON_EXISTING" in app.config)
print(app.config["NON_EXISTING"])
```

will give you

```
False
None
```

1. Having the following folder structure

<details>
<summary> Project structure </summary>

```13:57 $ tree -v
.
├── 1_no_dynaconf.py
└── 2_dynaconf.py

0 directories, 2 files
```
</details>

2. Having the following config files:

<details>
<summary> Config files </summary>
No config files used at all
</details>

3. Having the following app code:

<details>
<summary> Code </summary>

**1_no_dynaconf.py**
```python
from flask import Flask

app = Flask(__name__)
print("NON_EXISTING" in app.config)
print(app.config["NON_EXISTING"])
```

**2_dynaconf.py**
```python
from flask import Flask
from dynaconf import FlaskDynaconf

app = Flask(__name__)
FlaskDynaconf(app=app)
print("NON_EXISTING" in app.config)
print(app.config["NON_EXISTING"])
```
</details>

4. Executing under the following environment

<details>
<summary> Execution </summary>

used Python 3.8.5, Flask 1.1.2, dynaconf 3.1.2

```bash
# other commands and details?
# virtualenv activation?

$ python 1_no_dynaconf.py
$ python 2_dynaconf.py
```

</details>

**Expected behavior**
Queriing non existing key via `config[key]` should raise `KeyError`, not returning `None`.

**Environment (please complete the following information):**
 - OS: Linux/Kubuntu20.04
 - Dynaconf Version 3.1.2
 - Flask 1.1.2

**Additional context**

**Comment by rochacbruno** [comment]
Duplicate of #502 already merged and coming in the next release

**Comment by shsmad** [comment]
> Duplicate of #501 already merged in #502 and coming in the next release

Not quite duplicate. #501 and #502 are about `in` syntax, which is `__contains__` for. But my case is accessing `variable[key]`, so `DynaconfConfig.__getitem__` is used, which calls `DynaconfConfig.get` which calls `flask.config.Config.get` with default value set to `None`. But accessing real `flask.config.Config` via `[]` does **not** uses default value.

**Comment by rochacbruno** [comment]
@shsmad you are right, I just added #522 to fix this problem, the new release must come monday at morning. (I usually don't do releases on friday)
